app/frontend/components/Braintree/BraintreeCardFields.js

```diff
@@ -116,21 +115,42 @@ class BraintreeCardFields extends Component {
 
   submit(event?: SyntheticEvent) {
     if (event) event.preventDefault();
+    this.resetErrors();
 
     return new Promise((resolve, reject) => {
       if (!this.state.hostedFields) return reject();
 
       this.state.hostedFields.tokenize({
         vault: this.props.recurring,
       }, (error: ?BraintreeError, data: HostedFieldsTokenizePayload) => {
-        if (error) return reject(error);
+        if (error) {
+          this.processTokenizeErrors(error);
+          return reject(error);
+        }
         this.teardown();
         resolve(data);
-        console.log('Success BraintreeCardFields:', data);
       });
     });
   }
 
+  resetErrors() {
+    const newErrors = {};
+    Object.keys(this.state.errors).forEach((key) => {
+      newErrors[key] = false;
+    });
+    this.setState({errors: newErrors});
```

Comment:
I'm a bit confused by this - why not just set `errors: {}` if you want to reset the errors?